httputil/client.go

```diff
@@ -30,3 +39,52 @@ func GetRequestIP(req *http.Request) string {
 
 	return strings.ReplaceAll(addresses[0], " ", "")
 }
+
+func DefaultTransport() *http.Transport {
+	return &http.Transport{
+		DisableKeepAlives:   DefaultDisableKeepAlives,
+		MaxConnsPerHost:     DefaultMaxConnsPerHost,
+		MaxIdleConnsPerHost: DefaultMaxIdleConnsPerHost,
+		IdleConnTimeout:     DefaultIdleConnTimeout,
+	}
+}
+
+type HttpClientOptions func(*http.Client)
+
+// DefaultHttpClient returns a default HTTP client with a custom transport configuration.
+// It disables keep-alives, sets max connections per host, and configures idle connection timeout.
+// This is useful for clients that need to make many short-lived requests without reusing connections.
+// It also sets a default timeout of 30 seconds.
+func DefaultHttpClient() *http.Client {
+	return &http.Client{
+		Transport: DefaultTransport(),
+		Timeout:   DefaultRequestTimeout,
+	}
+}
+
+func NewHttpClient(options ...HttpClientOptions) *http.Client {
```

Comment:
I would find useful to have a 2nd constructor like `NewHttpClientFromConfig`:

```go
func NewHttpClientFromConfig(conf *config.Config) *http.Client {
```

Wdyt?

Reply:
Or we can pass an optional prefix for the conf keys like `router`. We can skip this if you think we won't need it. Worst case we'll do a PR later if we feel the need :+1: